src/dotty/tools/dotc/typer/Namer.scala

```diff
@@ -568,19 +569,9 @@ class Namer { typer: Typer =>
           val cls = typedAheadAnnotation(annotTree)
           val ann = Annotation.deferred(cls, implicit ctx => typedAnnotation(annotTree))
           denot.addAnnotation(ann)
-          if (cls == defn.InlineAnnot) {
-            hasInlineAnnot = true
-            addInlineInfo(denot, original)
-          }
-        }
-        if (!hasInlineAnnot && denot.is(InlineMethod)) {
-          // create a @inline annotation. Currently, the inlining trigger
-          // is really the annotation, not the flag. This is done so that
-          // we can still compile inline methods from Scala2x. Once we stop
-          // being compatible with Scala2 we should revise the logic to
-          // be based on the flag. Then creating a separate annotation becomes unnecessary.
-          denot.addAnnotation(Annotation(defn.InlineAnnot))
-          addInlineInfo(denot, original)
+          if (cls == defn.InlineAnnot && denot.is(Method, butNot = Accessor))
+            denot.setFlag(Inline)
+          if (denot.isInlineMethod) addInlineInfo(denot, original)
```

Comment:
This means that `addInlineInfo` will be called for every annotation if the method is marked inline, previously `hasInlineAnnot` prevented that.